### PR TITLE
chore: introduce phpcs and phpstan

### PR DESCRIPTION
**Pull request by Levdbas**
Related:

- #152
- #153 
- #154
- #155 
- #156
- #157
- #158 
- #159

## Issue
Since the creation of the starter theme a lot has happened in PHP, WordPress and Timber. I want to make the starter-theme a more modern starting point for custom theme development.

As mentioned by @szepeviktor in https://github.com/timber/starter-theme/issues/147#issuecomment-1805365281, it's a good idea to introduce a coding style.

I would like the help of you @szepeviktor  and @gchtr to help me enhance the introduction of these two tools. This is my first time introducing them in a project, so any help/tips are welcome.

I want to start linting file and introduce taskrunners when we have the overhaul with all the other PRs in place. So linting would be a final step as I see it now.

## Solution
Introduce phpcs and phpstan so we can start linting files.

## Impact
A coding style to adhere to is always great.

## Usage Changes
No

## Considerations
I just plucked some configurations from the internet and that can be improved of course, please help out 🗡️ 

## Testing
no

### CODE REVIEW

**Comment by Levdbas** [comment]
Hi @szepeviktor , especially looking for your feedback on this!

**Comment by szepeviktor** [comment]
@Levdbas What is the current error count?

**Comment by Levdbas** [comment]
> @Levdbas What is the current error count?

Currently 30 PHPstan issues but most of them will be fixed by #152 by the correct `use`. I now also was able to scan the files in the root directory by changing the config a bit.

**Comment by Levdbas** [comment]
@szepeviktor , Lukas and I discussed the PHP cs rulesets. Could you advice in how to implement https://github.com/szepeviktor/phpcs-psr-12-neutron-hybrid-ruleset while keep the possibility that users can add their own rules? I am pretty new to the whole ruleset creation thing so any help is appreciated!

**Comment by szepeviktor** [comment]
> I am pretty new to the whole ruleset creation thing so any help is appreciated!

A ruleset is simply one XML file with rules listed in it.

**Comment by szepeviktor** [comment]
One can disable an existing rule or add a new one in the theme.

**Comment by szepeviktor** [comment]
> Could you advice in how to implement

BTW This frankenstein is dangerous. It wears out your keyboard and your brain.

**Comment by Levdbas** [comment]
Updated the PHPCS ruleset base

File: phpcs.xml.dist [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	<exclude-pattern>*/vendor/*</exclude-pattern>
+   <exclude-pattern>*/resources/*</exclude-pattern>
+	<exclude-pattern>*/dist/*</exclude-pattern>
+	<exclude-pattern>*/tests/*</exclude-pattern>
+
+	<!-- Path to strip from the front of file paths inside reports (displays shorter paths) -->
+	<arg name="basepath" value="." />
+
+	<!-- Set a minimum PHP version for PHPCompatibility -->
+	<config name="testVersion" value="7.4-" />
```
Comment (gchtr) [inline phpcs.xml.dist]: Should we use PHP 8.1 here maybe, since we will require Timber 2.2.0, which requires PHP 8.1 as well?

Comment (Levdbas) [reply phpcs.xml.dist]: Yes, I will update the tests matrix as well to 8.1

**Review by gchtr — APPROVED** [review]

**Comment by gchtr** [comment]
@Levdbas Thanks for the update. I think it’s good to use a standard like the 10up standard. It means less work for us. And the phpcs.xml.dist looks a bit more accessible now.

In my opinion, it’s easy to spot that we load in a standard through Composer which is then applied in phpcs.xml.dist. This also makes it easy to swap it out with another standard of your choice.

**Pull request merged**